client-sdk/ts-web/core/src/hdkey.ts

```diff
@@ -0,0 +1,100 @@
+import {sha512} from 'js-sha512';
+import {SignKeyPair, sign} from 'tweetnacl';
+import {generateMnemonic, mnemonicToSeed, validateMnemonic} from 'bip39';
+import {concat} from './misc';
+
+const ED25519_CURVE = 'ed25519 seed';
+const HARDENED_OFFSET = 0x80000000;
+const pathRegex = new RegExp("^m(\\/[0-9]+')+$");
+
+/**
+ * HDKey handles hierarchical key generation according to ADR 0008
+ * https://github.com/oasisprotocol/oasis-core/blob/master/docs/adr/0008-standard-account-key-generation.md
+ */
+export class HDKey {
+    public readonly keypair: SignKeyPair;
+
+    /**
+     * Generates the keypair matching the supplied parameters
+     * @param mnemonic BIP-0039 Mnemonic
+     * @param index Account index
+     * @param passphrase Optional BIP-0039 passphrase
+     * @returns SignKeyPair for these parameters
+     */
+    public static async getAccountSigner(
+        mnemonic: string,
+        index: number = 0,
+        passphrase?: string,
+    ): Promise<SignKeyPair> {
+        if (index < 0 || index > 0x7fffffff) {
+            throw new Error('Account number must be >= 0 and <= 8388607');
```

Comment:
comparison checks against 2147483647, should the error message match?

Reply:
Good catch, missed a pair of `ff` !